clap_complete/src/dynamic/completer.rs

```diff
@@ -64,18 +65,81 @@ pub fn complete(
             if let Some(next_cmd) = current_cmd.find_subcommand(value) {
                 current_cmd = next_cmd;
                 pos_index = 1;
+                state = ParseState::ValueDone;
                 continue;
             }
         }
 
         if is_escaped {
             pos_index += 1;
+            state = ParseState::Pos(pos_index);
         } else if arg.is_escape() {
             is_escaped = true;
-        } else if let Some(_long) = arg.to_long() {
-        } else if let Some(_short) = arg.to_short() {
+            state = ParseState::ValueDone;
+        } else if let Some((flag, value)) = arg.to_long() {
+            if let Ok(flag) = flag {
+                let opt = current_cmd.get_arguments().find(|a| {
+                    a.get_long_and_visible_aliases()
+                        .map_or(false, |v| v.into_iter().find(|s| *s == flag).is_some())
+                });
+                state = opt.map_or_else(
```

Comment:
Looks like we don't have clippy checking this case.  We ban `map_or` in this code base.

Reply:
imo this is also too complex of a block to put in a combinator